### ISSUE TITLE
Tabbed UI Interface

### ISSUE THREAD

**Issue by mika76** [rubberduck-vba/Rubberduck#4892]
**Justification**
I would love this addin to create a tabbed interface for the editor. It's crazy irritating moving between windows in VB6 and tabs would really help.

**Description**
Instead of old-style MDI rather have tabs

**Additional context**
I found this old addin for VB6 called Code Help - it creates a tabbed interface using VB6 which is nice - maybe the idea/code could be salvaged and used in a more modern scenario: 
http://www.planet-source-code.com/vb/scripts/showcode.asp?txtCodeId=62468&lngWid=1

**Comment by MDoerner** [comment]
Thanks for suggesting this. It has already been discussed many times in the development chat, but I think we never got around to open an issue for it.

That being said, it will be a major undertaking to get this to work. Moreover, it might never work in break mode since we were unable so far to find a point to hook into the debugger.

**Comment by mika76** [comment]
Hey @MDoerner yeah I can imagine - that is why I added the example project I found - there seems to be a way they did it so maybe it could be useful 😊

**Comment by mansellan** [comment]
@mika76 Thanks for the find. We've had tabbed windows on our wishlist for a very long time.

I've taken a look at the source for the linked project. It works by injecting some pre-compiled assembly code into memory and then hooking it into certain window procedures. That's pretty awesome, but probably not a route we want to go down for RD.

Our current idea is to use [AvalonDock](https://github.com/xceedsoftware/wpftoolkit/wiki/AvalonDock) to provide not only tabbed code windows, but full-on docking as per modern Visual Studio.

**Comment by mika76** [comment]
Hey @mansellan 
Yeah a proper avalon dock environment would be the ultimate! The main frustration I have with the VB6 IDE these days is just that, the editor not having nice modern features and the IDE - the language is still good 😄 

But I think the source of the tab stuff is not pre-compiled but is in the Plugins\CHTabMDI2 folder...

If it's not the way you wanna go then no prob 😊 just thought it might be a step in the right direction...

**Comment by SteveLaycock** [comment]
Surely the approach more in line with RubberDuck would be to make use of the code explorer window and to add an option such as greying out or hiding modules that are not currently open for editing.

**Comment by retailcoder** [comment]
@SteveLaycock this issue is about enhancing the actual code panes though, i.e. turning the VBE's MDI (child windows) into a tear-tab UI like Visual Studio's, giving us the possibility to implement custom theming & syntax highlighting, code block folding, custom IntelliSense & autocompletion, etc.

**Comment by SteveLaycock** [comment]
Yes.  I understand that and I'm looking forward to it.  But as an interim solution.........

**Comment by bryceschober** [comment]
FWIW, I'm also using the [CodeSMART add-in](https://www.axtools.com/products-codesmart-vb6.php), which integrates its results panes as extra tabs in the VB6 IDE's "Immediates" pane.
